integrations/acquisition/covidcast/test_is_latest_issue.py

```diff
@@ -0,0 +1,250 @@
+    self._db.run_dbjobs()
+    
+    #check newer issue in signal_latest
+    sql = f'SELECT `issue` FROM {Database.latest_table} where `time_value` = 20200414  '
+    self._db._cursor.execute(sql)
+    record2 = self._db._cursor.fetchall()
+    self.assertEqual(record, record2) #same as previous as is_latest did not change
+
+    #dynamic check for signal_history's list of issue
+    self._db._cursor.execute(f'SELECT `issue` FROM {Database.history_table}')
+    record3 = self._db._cursor.fetchall()
+    self.assertEqual(len(record3),totalRows + 1) #ensure 3 added (1 of which refreshed)
+    self.assertEqual(20200416,max(record3)[0]) #max of the outputs is 20200416 , extracting from tuple
+    
+    #check older issue not inside latest, empty field
+    sql = f'SELECT * FROM {Database.latest_table} where `time_value` = 20200414 and `issue` = 20200414 '
+    self._db._cursor.execute(sql)
+    emptyRecord = list(self._db._cursor.fetchall())
+    empty = []
+    self.assertEqual(empty, emptyRecord)
+      
+  # We want to test src_sig to make sure rows are added to *_dim only when needed
+    #new src, sig (ensure that it is added into *_dim)
+    #old src, sig (ensure that it is NOT added into *_dim)
+    #new geo (added)
+    #old geo (not added)
+  def test_src_sig(self):
+    #BASE CASES
+    rows = [
+      CovidcastRow('src', 'sig', 'day', 'state', 20200414, 'pa', 
+      2, 2, 2, nmv, nmv, nmv, 20200414, 0),
+      CovidcastRow('src', 'sig', 'day', 'county', 20200414, '11111',
+      3, 3, 3, nmv, nmv, nmv, 20200414, 0)
+    ]
+    self._db.insert_or_update_bulk(rows)
+    self._db.run_dbjobs()
+    self._db._cursor.execute(self.viewGeoDim)
+    record = self._db._cursor.fetchall()
+    geoDimRows = len(list(record))
+
+    self._db._cursor.execute(self.viewSignalDim)
+    record = self._db._cursor.fetchall()
+    sigDimRows = len(list(record))
+
+    self._db._cursor.execute(self.viewSignalLatest)
+    record = self._db._cursor.fetchall()
+    sigLatestRows = len(list(record))
+
+    #test not added first
+    with self.subTest(name='older src and sig not added into sig_dim'):
+      oldSrcSig = [
+        CovidcastRow('src', 'sig', 'day', 'state', 20211111, 'pa', #new src, new sig
+                    99, 99, 99, nmv, nmv, nmv, 20211111, 1),
+        CovidcastRow('src', 'sig', 'day', 'county', 20211111, '11111', #new src, new sig, same geo
+                    99, 99, 99, nmv, nmv, nmv, 20211111, 1)
+          ]  
+      self._db.insert_or_update_bulk(oldSrcSig)
+      self._db.run_dbjobs()
+
+      #testing src, sig 
+      self._db._cursor.execute(self.viewSignalDim)
+      record = self._db._cursor.fetchall()
+      res = [('src','sig')] #output
+      self.assertEqual(res , list(record))
+
+      #ensure new entries are added in latest
+      self._db._cursor.execute(self.viewSignalLatest)
+      record = self._db._cursor.fetchall()
+      self.assertEqual(len(list(record)), sigLatestRows + 2) #2 original, 2 added
```

Comment:
A pattern like this will be easier to maintain -- we'll be able to add new tests without having to update all subsequent expressions.
```suggestion
      record_length = len(list(record))
      self.assertEqual(record_length, sigLatestRows + 2) #2 original, 2 added
      sigLatestRows = record_length
```